src/Plugin.php

```diff
@@ -208,6 +211,9 @@ public static function init() {
 
     // Removes the suffix '/page/1' from archive URLs.
     add_filter('paginate_links', __CLASS__  . '::paginate_links');
+
+    // Front-end output of custom page title.
+    add_filter('woocommerce_page_title', __CLASS__.'::woocommerce_page_title');
```

Comment:
Did you try wrapping this into `is_term()`?

Reply:
is_term is deprecated and term_exists leads to a false. Not sure if I should use the optional params?

Reply:
1. The callback should remain in the feature class, so that it is clear to which feature this hook registration belongs.
2. There should be a space before and after the string concatenation operator.
```suggestion
    add_filter('woocommerce_page_title', __NAMESPACE__ . '\ProductAttributePageTitle::woocommerce_page_title');
```